uisdk/src/test/java/tech/dojo/pay/uisdk/presentation/ui/carddetailscheckout/mapper/CardCheckOutFullCardPaymentPayloadMapperTest.kt

```diff
@@ -0,0 +1,392 @@
+package tech.dojo.pay.uisdk.presentation.ui.carddetailscheckout.mapper
+
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.whenever
+import tech.dojo.pay.sdk.card.entities.DojoAddressDetails
+import tech.dojo.pay.sdk.card.entities.DojoCardDetails
+import tech.dojo.pay.sdk.card.entities.DojoCardPaymentPayLoad
+import tech.dojo.pay.uisdk.presentation.ui.carddetailscheckout.entity.SupportedCountriesViewEntity
+import tech.dojo.pay.uisdk.presentation.ui.carddetailscheckout.state.CardDetailsCheckoutState
+import tech.dojo.pay.uisdk.presentation.ui.carddetailscheckout.state.CheckBoxItem
+import tech.dojo.pay.uisdk.presentation.ui.carddetailscheckout.state.InputFieldState
+
+class CardCheckOutFullCardPaymentPayloadMapperTest {
+    private val isStartDestination: Boolean = false
+    private lateinit var mapper: CardCheckOutFullCardPaymentPayloadMapper
+
+    @Before
+    fun setUp() {
+        mapper = CardCheckOutFullCardPaymentPayloadMapper()
+    }
+
+    @Test
+    fun `given calling getPaymentPayLoad with email and billing enabled then should returns correct payload with email and billing address`() {
+        // Arrange
+        val currentState: CardDetailsCheckoutState = mock()
+        whenever(currentState.isEmailInputFieldRequired).thenReturn(true)
+        whenever(currentState.emailInputField).thenReturn(InputFieldState("test@example.com"))
+        whenever(currentState.isBillingCountryFieldRequired).thenReturn(true)
+        whenever(currentState.isPostalCodeFieldRequired).thenReturn(true)
+        whenever(currentState.currentSelectedCountry).thenReturn(
+            SupportedCountriesViewEntity(
+                countryCode = "US",
+                countryName = "USA",
+                isPostalCodeEnabled = false,
+            ),
+        )
+        whenever(currentState.postalCodeField).thenReturn(InputFieldState("12345"))
+        whenever(currentState.checkBoxItem).thenReturn(
+            CheckBoxItem(
+                isVisible = false,
+                isChecked = true,
+                messageText = "",
+            ),
+        )
+        whenever(currentState.cardNumberInputField).thenReturn(InputFieldState("1234567812345678"))
+        whenever(currentState.cardHolderInputField).thenReturn(InputFieldState("John Doe"))
+        whenever(currentState.cardExpireDateInputField).thenReturn(InputFieldState("1225"))
+        whenever(currentState.cvvInputFieldState).thenReturn(InputFieldState("123"))
+        whenever(currentState.checkBoxItem).thenReturn(
+            CheckBoxItem(
+                isVisible = true,
+                isChecked = true,
+                messageText = "",
+            ),
+        )
```

Comment:
```suggestion
        val currentState: CardDetailsCheckoutState = mock {
          on { isEmailInputFieldRequired } doReturn true
          on { emailInputField } doReturn InputFieldState("test@example.com")
          ....
       }
```
This way would look cleaner in my opinion. `currentState.` would be repeated for example.